src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -89,7 +89,14 @@ protected override IRazorPresentationParams CreateRazorRequestParameters(UriPres
         if (!fileName.EndsWith(".razor", FilePathComparison.Instance))
         {
             _logger.LogInformation("Last file in the drop was not a single razor file URI.");
-            return null;
+            razorFileUri = request.Uris
+                .Where(x => Path.GetFileName(x.GetAbsoluteOrUNCPath()).EndsWith(".razor", FilePathComparison.Instance))
+                .LastOrDefault();
```

Comment:
Why do we check the last file is a .razor file, only the then check if any file is a .razor file if that fails? Seems redundant.

Reply:
Didn't want to cause regressions to existing behavior in case this was done intentionally

Reply:
Thats a noble goal, but I think in this case there probably isn't anything worth preserving. If the existing behaviour was important in some way, the bug wouldn't exist